### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by Lucas-Masaba**
# Requirements ⚙️ 
- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical questions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,11 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT *  FROM animals WHERE neutered;
 SELECT *  FROM animals WHERE name NOT LIKE 'Gabumon';
-SELECT *  FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT *  FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+SELECT COUNT(*) FROM animals;
+SELECT * FROM animals WHERE escape_attempts = 0;
```
Comment (Meltrust) [inline queries.sql]: - [x] This one should count the animals that have tried to escape, not list them.  Example:

`   SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;`

Kindly correct that one. 💪

File: data.sql [hunk 2]
```diff
@@ -3,4 +3,31 @@
...
+DELETE * FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```
Comment (Meltrust) [inline data.sql]: - [ ] According to the project requirements:

image

Therefore, please verify the transactions when required.  This means using a SELECT statement.

According to said requirements, please, kindly use a SELECT statement after each commit, rollback, and update.  That way you can verify the flow of said transactions, updates, and rollbacks.

Example:

```SQL
  
  BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made


  (etc...)

```

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Lucas-Masaba,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice code indentation and organization ✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Lucas-Masaba) [reply queries.sql]: Thank you for that observation. I have successfully worked on that

File: data.sql (hunk 2 above)
Comment (Lucas-Masaba) [reply data.sql]: Thank you again for your observation. I have done all that is required.

**Review by Meltrust — APPROVED** [review]
Hi @Lucas-Masaba,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**